File: Lecture16-Bluring/main.cpp
#include <opencv4/opencv2/opencv.hpp>

using namespace cv;

void blur() {
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/kask.jpg");
    Mat blr;
    namedWindow("Original", WINDOW_AUTOSIZE);
    namedWindow("Blur", WINDOW_AUTOSIZE);

    blur(img, blr, Size(8, 9));
    imshow("Original", img);
    imshow("Blur", blr);
    imwrite("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/blur.jpg", blr);
    waitKey(0);
    destroyAllWindows();
}

void gaussianBlur()
{
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/kask.jpg");
    Mat blr;
    namedWindow("Original", WINDOW_AUTOSIZE);
    namedWindow("Gaussian Blur", WINDOW_AUTOSIZE);

    GaussianBlur(img, blr, Size(11, 11), 3, 3);
    imshow("Original", img);
    imshow("Gaussian Blur", blr);
    imwrite("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/Gaussian blur.jpg", blr);

    waitKey(0);
    destroyAllWindows();
}

void BoxFilter()
{
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/kask.jpg");
    Mat blr;
    namedWindow("Original", WINDOW_AUTOSIZE);
    namedWindow("Box Filter", WINDOW_AUTOSIZE);

    boxFilter(img, blr, -1, Size(7, 7));
    imshow("Original", img);
    imshow("Box Filter", blr);
    imwrite("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/Box Filter.jpg", blr);

    waitKey(0);
    destroyAllWindows();
}

void MedianBlur()
{
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/kask.jpg");
    Mat blr;
    namedWindow("Original", WINDOW_AUTOSIZE);
    namedWindow("Median Blur", WINDOW_AUTOSIZE);

    medianBlur(img, blr, 9);
    imshow("Original", img);
    imshow("Median Blur", blr);
    imwrite("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/Median Blur.jpg", blr);

    waitKey(0);
    destroyAllWindows();
}

void bilateralFilter()
{
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders16_Bulaniklastirma/kask.jpg");
    Mat blr;
    namedWindow("Original", WINDOW_AUTOSIZE);
    namedWindow("Bilateral filter", WINDOW_AUTOSIZE);

    bilateralFilter(img, blr, -1, 50, 9);
    imshow("Original", img);
    imshow("Bilateral filter", blr);

    waitKey(0);
    destroyAllWindows();
}

int main() {
    //blur();
    //gaussianBlur();
    //BoxFilter();
    //MedianBlur();
    //bilateralFilter();
    return 0;
}

File: Lecture16-Bluring/main.py
import cv2 as cv

img = cv.imread("fire.jpg")
cv.namedWindow("Orijinal", cv.WINDOW_AUTOSIZE)
cv.imshow("Orijinal", img)

def blur():
    cv.namedWindow("Blur", cv.WINDOW_AUTOSIZE)
    blurImage = cv.blur(img, (3,3))
    cv.imshow("Blur", blurImage)
    cv.imwrite("FireBlur.jpg", blurImage)


def gaussianBlur():
    cv.namedWindow("Gaussian Blur", cv.WINDOW_AUTOSIZE)
    gaussianImg = cv.GaussianBlur(img, (5,5), 0)
    cv.imshow("Gaussian Blur", gaussianImg)
    cv.imwrite("FireGaussianBlur.jpg", gaussianImg)

def BoxFilter():
    cv.namedWindow("Box Filter", cv.WINDOW_AUTOSIZE)
    boxImg = cv.boxFilter(img, -1, (5,5))
    cv.imshow("Box Filter", boxImg)
    cv.imwrite("BoxFilterImg.jpg", boxImg)

def medianBlur():
    cv.namedWindow("Median Blur", cv.WINDOW_AUTOSIZE)
    medianImg = cv.medianBlur(img, 5)
    cv.imshow("Median Blur", medianImg)
    cv.imwrite("MedianImg.jpg", medianImg)

def BilateralFiltre():
    cv.namedWindow("Bilateral Filtre", cv.WINDOW_AUTOSIZE)
    bilateralImg = cv.bilateralFilter(img, 9, 75, 75)
    cv.imshow("Bilateral Filtre", bilateralImg)
    cv.imwrite("BilateralImg.jpg", bilateralImg)

if __name__ == '__main__':
    BilateralFiltre()
    cv.waitKey(0)
    cv.destroyAllWindows()